src/dune_engine/build_system.ml

```diff
@@ -337,6 +337,7 @@ type t =
   ; sandboxing_preference : Sandbox_mode.t list
   ; mutable rule_done : int
   ; mutable rule_total : int
+  ; mutable rule_errors : Exn_with_backtrace.t list
```

Comment:
I'd call this just `errors`, since this includes all kinds of errors, not just ones raised during rule execution.